src/executables/util_cs.c

```diff
@@ -3296,17 +3328,41 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
 	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
 				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
+	  if (error != NO_ERROR)
+	    {
+	      goto print_applyinfo_usage;
+	    }
 	}
 
     check_applied_info_end:
       if (error != NO_ERROR)
 	{
 	  PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	  goto print_applyinfo_usage;
 	}
       error = NO_ERROR;
 
       if (check_master_info)
 	{
+	  check_master_info = false;
+	  for (; ha_node_list_pp[ha_node_cnt] != NULL;)
```

Comment:
It seems that `ha_node_cnt` is not a cnt, but index. You're not counting node, aren't you?

Reply:
In opinion, ha_node_idx is better to read the code. do you agree?

Reply:
seems better